core/src/dht/dht_store.rs

```diff
@@ -59,22 +59,31 @@ impl DhtStore {
     pub fn get_links(
         &self,
         address: Address,
-        tag: String,
+        link_type: String,
+        tag: Option<String>,
     ) -> Result<BTreeSet<EntityAttributeValueIndex>, HolochainError> {
         let filtered = self.meta_storage.read()?.fetch_eavi(&EaviQuery::new(
             Some(address).into(),
-            EavFilter::multiple(vec![
-                Attribute::LinkTag(tag.clone()),
-                Attribute::RemovedLink(tag),
-            ]),
+            EavFilter::predicate(move |attr| {
+                match attr {
+                    Attribute::LinkTag(query_link_type, query_tag) | Attribute::RemovedLink(query_link_type, query_tag) => {
```

Comment:
Highlight: This is where filtering is done based on if a tag is provided (so match both tag and type) or only a type is provided (return all tags of that type)